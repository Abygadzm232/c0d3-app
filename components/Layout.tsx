import React from 'react'
import AppNav from './AppNav'
import Footer from './Footer'
import SessionContext from '../helpers/contexts/session'
import _ from 'lodash'

type Props = {
  children: React.ReactElement
}

const Layout: React.FC<Props> = ({ children }) => {
  const { session } = React.useContext(SessionContext)

  if (session && session.user) {
    return (
      <>
        <AppNav username={session.user.username} loggedIn />
        <div className="container">{children}</div>
        <Footer />
      </>
    )
  }

  return (
    <>
      <AppNav />
      <div className="container">{children}</div>
      <Footer />
    </>
  )
}

export default Layout
