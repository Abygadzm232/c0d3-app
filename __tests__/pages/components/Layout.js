import React from 'react'
import renderer from 'react-test-renderer'
import Layout from '../../../components/Layout'

describe('Snapshot for Layout', () => {
  test('With Layout', () => {
    const component = renderer.create(
      <Layout>
        <div className="row" style={{ height: '100vh' }}>
          <div className="card col-6">
            <div className="card-body">
              <h5 className="card-title">Card title</h5>
              <p className="card-text">
                Some quick example text to build on the card title and content.
              </p>
              <a href="#" className="btn btn-primary">
                Go somewhere
              </a>
            </div>
          </div>
        </div>
      </Layout>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
