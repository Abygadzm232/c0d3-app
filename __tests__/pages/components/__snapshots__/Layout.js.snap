// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for Layout With Layout 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-light justify-content-between bg-white"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand text-primary font-weight-bold"
        href="#"
      >
        C0D3
      </a>
      <div
        id="navbarNav"
      >
        <div
          className="navbar-nav collapse navbar-collapse"
        >
          <a
            className="nav-item nav-link active"
            href="#"
          >
            Home 
            <span
              className="sr-only"
            >
              (current)
            </span>
          </a>
          <a
            className="nav-item nav-link"
            href="#learning"
          >
            Learning Process
          </a>
          <a
            className="nav-item nav-link"
            href="#journey"
          >
            Journey
          </a>
          <a
            className="nav-item nav-link"
            href="https://chat.c0d3.com"
          >
            Help
          </a>
        </div>
      </div>
      <div>
        <a
          className="btn border mr-3"
          href="/login"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Login
        </a>
      </div>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div
      className="row"
      style={
        Object {
          "height": "100vh",
        }
      }
    >
      <div
        className="card col-6"
      >
        <div
          className="card-body"
        >
          <h5
            className="card-title"
          >
            Card title
          </h5>
          <p
            className="card-text"
          >
            Some quick example text to build on the card title and content.
          </p>
          <a
            className="btn btn-primary"
            href="#"
          >
            Go somewhere
          </a>
        </div>
      </div>
    </div>
  </div>,
  <footer
    className="mt-5 font-weight-light text-center pb-5 text-secondary"
  >
    Â© Copyright 2020 GarageScript
    <br />
    All Rights Reserved
  </footer>,
]
`;
